### PR TITLE
Task 4: Submission

### PR DESCRIPTION
**Pull request by redshadowhero**
This is my submission.

A few notes:
* The Docker build folder, systemd service file, docker repo file (for yum), and the bash script that echos the date are in `playbooks/roles/docker/files`
* This has been tested 'end to end' on a fresh Centos 7 install; most of the documentation was written from this test. It's also been tested on Mac OS X Sierra and Debian 7, but those machines have already been set up for development, so I'm not 100% sure if any extra development files are needed other than the ones noted in `INSTALL.md`

Please let me know if you have any questions!

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+You'll need to have some commands available to start. Consult your local package manager if they aren't:
+
+1. python 2.7
+2. pip
+3. virtualenv
+4. gcc
+5. rsync
+
```
Comment (kmullican) [inline INSTALL.md:18]: dos2unix

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+There are minimal instructions for centos 7 here that should get you going fairy quickly, execute these commands as root to bootstrap the machine quickly:
+```
+yum -y install http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+yum -y update
+yum -y install python-pip gcc python-devel openssl-devel rsync
+pip install virtualenv
+```
+
+Also, git mangles the permission on the ssh keys, correct them with: `chmod 0600 misc/hudsonalpha*`
+
```
Comment (kmullican) [inline INSTALL.md:33]: Be sure all files are in unix format by running the following command (not doing so causes errors when executing do/setup.py, among other files)

find /path/to/code -type f -exec dos2unix {} \;

File: playbooks/roles/docker/files/image/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Very simple docker service to output the time every 10 seconds
+
+FROM centos:7
+MAINTAINER "Chris King" <tchrisking@gmail.com>
+
+COPY datetime /usr/local/bin/datetime
+
```
Comment (kmullican) [inline playbooks/roles/docker/files/image/Dockerfile:8]: RUN chmod +x /usr/local/bin/datetime

File: playbooks/roles/docker/files/datetime.service [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+Requires=docker.service
+
+[Service]
+User=wannabe
+TimeoutStartSec=0
+Restart=always
+ExecStartPre=-/usr/bin/docker stop datetime
+ExecStartPre=-/usr/bin/docker rm datetime
+ExecStart=/usr/bin/docker run -d --name datetime datetime:habuild
+ExecStopPre=-/usr/bin/docker stop -t 11 datetime
```
Comment (kmullican) [inline playbooks/roles/docker/files/datetime.service]: ExecStopPre causes error when running this service -- ExecStop (without the "Pre") runs OK

**Review by kmullican — COMMENTED** [review]
Nice job!  I tested on Fedora, just for fun

**Comment by redshadowhero** [comment]
I fixed the noted issues in 092c7daac960ed4b66f7daf9a60a697f7a3180cf

Nice catch on the service file -- sometimes I'll yank a line in vim and paste it elsewhere, changing one thing while leaving stuff tacked in that I really shouldn't have.

I also fired up a Fedora 25 (Workstation edition, but I don't think it matters) for this test, but I didn't encounter the dos encoding issue. Still, it's probably better to be verbose, so I updated the corresponding documentation.

**Pull request closed without merging**